lib/sourcemaps.js

```diff
@@ -71,27 +73,28 @@ exports.concatenateSourceMaps = function(sourceFilename, mapsWithOffsets, source
 
   // convert from library internals format to canonical
   var normalized = JSON.parse(JSON.stringify(generated));
+  var sourcePath;
 
   if (includeSourcesContent) {
     normalized.sourcesContent = normalized.sources.map(function(source) {
     //if (contentsBySource[source])
       //return contentsBySource[source];
 
-    if (source.substr(0, 6) === 'file:/')
-      return fs.readFileSync(source.replace(/^file:\/+/, '/')).toString();
+    sourcePath = filePath(source);
```

Comment:
We should probably handle relative sources here? Ie, if it starts with `x.js` or `./x.js` then it should be taken to be relative to sourceFileName?